app/views/alchemy/admin/resources/_form.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= alchemy_form_for resource_instance_variable, url: resource_path(resource_instance_variable) do |f| %>
+<%= alchemy_form_for resource_instance_variable, url: resource_path(resource_instance_variable, q: params[:q]) do |f| %>
```

Comment:
This is a POST form, does this work with GET parameters? Normally you would put this into a hidden field. Especially because `q` can be a rather large object and url's have limited length.